src/core/tests/type_prop/squeeze.cpp

```diff
@@ -145,7 +280,12 @@ TEST_P(SqueezeTest, partial_shape_dimension_propagation_parameter_axes_no_data)
     const auto squeeze = std::make_shared<op::v0::Squeeze>(param, axes_node);
 
     EXPECT_EQ(squeeze->get_element_type(), element::f32);
-    EXPECT_EQ(squeeze->get_output_partial_shape(0), PartialShape::dynamic());
+
+    if (axes.size() == 0 || param->get_output_partial_shape(0).rank().is_dynamic()) {
+        EXPECT_EQ(squeeze->get_output_partial_shape(0), PartialShape::dynamic());
```

Comment:
I think is not OK the the correct shape should be set in test parameter and stored in variable `exp_shape`
It looks like bypass expected result.
Update the `static_partial_shapes_test_values ` and/or `empty_axes_test_values `.

Reply:
> This line should be restored to EXPECT_EQ(squeeze->get_output_partial_shape(0), exp_shape);

This line can't be "restored" to such check as it wasn't originally like that. 
The original check expected for all cases fully dynamic shape instead of "exp_shape":
`EXPECT_EQ(squeeze->get_output_partial_shape(0), PartialShape::dynamic());`

As the goal of the improvement is to make the output rank not always dynamic, the original check is no longer universal as sometimes the rank can be static.
So I have updated the check to check compatibility of output shape with "exp_shape" (expected for constant input). 
`EXPECT_TRUE(squeeze->get_output_partial_shape(0).compatible(exp_shape));`

Tests expected specific shape/rank have been added separately.